File: PdfTemplateEditor/PdfTemplateEditor/Common/RectangleNormal.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AxiomCoders.PdfTemplateEditor.Common
{
    public class RectangleNormal
    {
        public float top;
        public float left;
        public float bottom;
        public float right;

        public RectangleNormal()
        {
        }


        public bool Intersect(RectangleNormal rect)
        {
			return !((this.left - rect.right >= -0.001f)
			|| (rect.left - this.right >= -0.001f)
			|| (this.top - rect.bottom >= -0.001f)
			|| (rect.top - this.bottom >= -0.001f));

            /*return !(this.left >= rect.right
                || rect.left >= this.right
                || this.top >= rect.bottom
                || rect.top >= this.bottom);*/
        }


        public bool Contains(RectangleNormal rect)
        {
			return ((rect.left - this.left >= -0.001f)
			&& (this.right - rect.right >= -0.001f)
			&& (rect.top - this.top >= -0.001f)
			&& (this.bottom - rect.bottom >= -0.001f));

            /*return (rect.left >= this.left
                && rect.right <= this.right
                && rect.top >= this.top
                && rect.bottom <= this.bottom);*/
        }



        public void MoveTo(float locX, float locY)
        {
            float w, h;
            w = this.right - this.left;
            h = this.bottom - this.top;

            this.left = locX;
            this.top = locY;

            this.right = (this.left + w);
            this.bottom = (this.top + h);		
        }
    }
}

File: PdfFactory/source/RectangleNormal.c
#include "RectangleNormal.h"
#include "MemoryManager.h"

DLLEXPORT_TEST_FUNCTION struct RectangleNormal* RectangleNormal_Create(float x, float y, float width, float height)
{
	struct RectangleNormal *xRect;
	xRect = (struct RectangleNormal*)MemoryManager_Alloc(sizeof(struct RectangleNormal));
	RectangleNormal_Init(xRect, x, y, width, height);
	return xRect;
}


DLLEXPORT_TEST_FUNCTION void RectangleNormal_Init(struct RectangleNormal *self, float x, float y, float width, float height)
{
	self->top = y;
	self->left = x;
	self->bottom = self->top + height;
	self->right = self->left + width;
}



DLLEXPORT_TEST_FUNCTION void RectangleNormal_Destroy(struct RectangleNormal *self)
{
	MemoryManager_Free(self);
}


DLLEXPORT_TEST_FUNCTION int RectangleNormal_Intersect(struct RectangleNormal *self, struct RectangleNormal *rect)
{
	return !((self->left - rect->right >= -0.001f)
		|| (rect->left - self->right >= -0.001f)
		|| (self->top - rect->bottom >= -0.001f)
		|| (rect->top - self->bottom >= -0.001f));

	/*return ! ( self->left >= rect->right
		|| rect->left >= self->right
		|| self->top >= rect->bottom
		|| rect->top >= self->bottom);*/
}


DLLEXPORT_TEST_FUNCTION int RectangleNormal_Contains(struct RectangleNormal *self, struct RectangleNormal *rect)
{
	return ((rect->left - self->left >= -0.001f)
		&& (self->right - rect->right >= -0.001f)
		&& (rect->top - self->top >= -0.001f)
		&& (self->bottom - rect->bottom >= -0.001f));

	/*return (rect->left >= self->left 
		&& rect->right <= self->right 
		&& rect->top >= self->top 
		&& rect->bottom <= self->bottom);*/
}



DLLEXPORT_TEST_FUNCTION void RectangleNormal_MoveTo(struct RectangleNormal *self, float locX, float locY)
{
	float w, h;
	w = self->right - self->left;
	h = self->bottom - self->top;

	self->left = locX;
	self->top = locY;

	self->right = self->left + w;
	self->bottom = self->top + h;	
}